### PR TITLE
ci: set up GitHub actions workflows

### LINKED ISSUE #13: meta: set up CI build
It'd be good to set up GitHub actions to do some basic sanity checks. I'd be happy to send a PR.

### PR DESCRIPTION
**Pull request by mdlayher**
Signed-off-by: Matt Layher <mdlayher@gmail.com>

Mostly copied from tailscale repo. I think this will work, but this is my first time setting up Actions.

Fixes #13.

### CODE REVIEW

**Comment by crawshaw** [comment]
These staticcheck results are not painting a very nice picture of staticcheck. They're all bogus. How about leaving it out for now and pushing ahead with the linux tests? This PR otherwise seems to work great.

**Comment by mdlayher** [comment]
Yep seems fine.

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Set up Go 1.14.2
```
Comment (bradfitz) [inline .github/workflows/linux.yml:16]: "Set up Go" ?

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.14.2
```
Comment (bradfitz) [inline .github/workflows/linux.yml:19]: IIRC, just saying "1.14" means the latest Go 1.14.x and not 1.14 exactly.

I'd rather not have to keep this in sync.

**Comment by mdlayher** [comment]
Done and done.

**Pull request merged**